### PR TITLE
[FIX] website: restrict access to optimize SEO

### PR DESCRIPTION
**Pull request by prec-odoo**
When user creates a new 'livechat channel' in website, and click on 'Optimize SEO' traceback will appear.

To reproduce the issue:

- Install 'website' and 'im_livechat' module
- Go to 'website' module
- Click on 'NEW' and then create a new 'Livechat Widget'
- Go to 'Site' and click on 'Optimize SEO'

Traceback in sentry-
```
ValueError: Invalid field 'website_meta_title' on model 'im_livechat.channel'
  File "odoo/http.py", line 2114, in __call__
    response = request._serve_db()
  File "odoo/http.py", line 1699, in _serve_db
    return service_model.retrying(self._serve_ir_http, self.env)
  File "odoo/service/model.py", line 133, in retrying
    result = func()
  File "odoo/http.py", line 1726, in _serve_ir_http
    response = self.dispatcher.dispatch(rule.endpoint, args)
  File "odoo/http.py", line 1921, in dispatch
    result = self.request.registry['ir.http']._dispatch(endpoint)
  File "addons/website/models/ir_http.py", line 234, in _dispatch
    response = super()._dispatch(endpoint)
  File "odoo/addons/base/models/ir_http.py", line 190, in _dispatch
    result = endpoint(**request.params)
  File "odoo/http.py", line 716, in route_wrapper
    result = endpoint(self, *args, **params_ok)
  File "addons/website/controllers/main.py", line 699, in get_seo_data
    res = record.read(fields)[0]
  File "odoo/models.py", line 3176, in read
    self.fetch(fields)
  File "odoo/models.py", line 3407, in fetch
    fields_to_fetch = self._determine_fields_to_fetch(field_names, ignore_when_in_cache=True)
  File "odoo/models.py", line 3474, in _determine_fields_to_fetch
    raise ValueError(f"Invalid field {field_name!r} on model {self._name!r}")
```

When user create a new livechat channel in website module, the 'get_seo_data' method is called and the value of 'res_model' is passed as 'im_livechat.channel'. Fixed this issue by raising UserError when user tries to access 'Optimize SEO'.

sentry-4241511888

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by prec-odoo** [comment]
Hello @odoo/sentry_reviewers ,

Please check this PR.

Thank You!

File: addons/website/controllers/main.py [hunk 1]
```diff
@@ -678,6 +678,8 @@ def get_seo_data(self, res_id, res_model):
             fields.extend(['website_indexed', 'website_id'])
 
         record = request.env[res_model].browse(res_id)
+        if res_model != 'website.page' and 'website.seo.metadata' not in request.env[res_model]._inherit_module.keys():
+            raise UserError(_("You can not optimize SEO for livechat."))
```
Comment (yelizariev) [inline addons/website/controllers/main.py:682]: The message says *for livechat*, but it's only one of possible scenarios, no?

**Review by yelizariev — CHANGES_REQUESTED** [review]

File: addons/website/controllers/main.py (hunk 1 above)
Comment (prec-odoo) [reply addons/website/controllers/main.py]: > The message says _for livechat_, but it's only one of possible scenarios, no?

@yelizariev , changes has been done.

**Review by rdeodoo — CHANGES_REQUESTED** [review]
Please never fix consequences of a bug, but fix instead the root cause.
Here, the problem is that the model receive the Edit Seo menu while it has no seo capability on that model (mixin).
The menu shouldn't even be there in the first place.
The website menu condition are done somewhere around here: https://github.com/odoo/odoo/blame/a0f6bcb046ca66737099fa7f211b345e32241cf5/addons/website/static/src/services/website_custom_menus.js#L86-L91
```js
registry.category('website_custom_menus').add('website.menu_optimize_seo', {
    Component: OptimizeSEODialog,
    isDisplayed: (env) => env.services.website.currentWebsite
        && env.services.website.isDesigner
        && !!env.services.website.currentWebsite.metadata.mainObject,
});
```
It should also be conditioned based on the fact that the model is "optimize seo" friendly.

**Comment by loco-odoo** [comment]
Fixed in https://github.com/odoo/odoo/pull/127043

**Pull request closed without merging**